python/lsst/pipe/tasks/setPrimaryFlags.py

```diff
@@ -20,15 +20,172 @@
+    tractInfo : `lsst.skymap.TractInfo`
+        Tract object
+    skyMap : `lsst.skymap.BaseSkyMap`
+        Sky tessellation object
+
+    Returns
+    -------
+    isTractInner : array-like of `bool`
+        True if the skyMap.findTract method returns
+        the same tract as tractInfo.
+    """
+    tractId = tractInfo.getId()
+    isTractInner = np.array([skyMap.findTract(s.getCoord()).getId() == tractId for s in sources])
+    return isTractInner
+
+
+def getPseudoSources(sources, pseudoFilterList, schema, log):
+    """Get a flag that marks pseudo sources.
+
+    Some categories of sources, for example sky objects,
+    are not really detected sources and should not be considered primary
+    sources.
+
+    Parameters
+    ----------
+    sources : `lsst.afw.table.SourceCatalog`
+        The catalog of sources for which to identify "pseudo"
+        (e.g. sky) objects.
+    pseudoFilterList : `list` of `str`
+        Names of filters which should never be primary
+
+    Returns
+    -------
+    isPseudo : array-like of `bool`
+        True for each source that is a pseudo source.
+        Note: to remove pseudo sources use `~isPseudo`.
+    """
+    # Filter out sources that should never be primary
+    isPseudo = np.zeros(len(sources), dtype=bool)
+    for filt in pseudoFilterList:
+        try:
+            pseudoFilterKey = schema.find("merge_peak_%s" % filt).getKey()
+            isPseudo |= sources[pseudoFilterKey]
+        except KeyError:
+            log.warn("merge_peak is not set for pseudo-filter %s" % filt)
+    return isPseudo
+
+
+def getDeblendPrimaryFlags(sources):
+    """Get flags generated by the deblender
+
+    scarlet is different than meas_deblender in that it is not
+    (necessarily) flux conserving. For consistency in scarlet,
+    all of the parents with only a single child (isolated sources)
+    need to be deblended. This creates a question: which type
+    of isolated source should we make measurements on, the
+    undeblended "parent" or the deblended child?
+    For that reason we distinguish between a DeblendedPrimary,
+    which is a source that has no children and uses the
+    isolated parents, and a DeblendedModelPrimary, which uses
+    the scarlet models for both isolated and blended sources.
+    In the case of meas_deblender, DeblendedModelPrimary is
+    `None` because it is not contained in the output catalog.
+
+    Parameters
+    ----------
+    sources : `lsst.afw.table.SourceCatalog`
+        A sourceCatalog that has already been deblended using
+        either meas_extensions_scarlet or meas_deblender.
+
+    Returns
+    -------
+    isBlended : array-like of `bool`
+        True for each source that was deblended from a parent with more
+        than one child. This includes blended sources that have children of
+        their own.
+    isIsolated : array-like of `bool`
+        True for isolated sources, regardless of whether or not they
+        were deblended.
```

Comment:
Looking back at the docs I don't know how to be more clear, so perhaps you can help me trim down this long explanation into something that can fit into the docs.

top level parents with multiple peaks: not contained in either `isBlended` or `isIsolated`.
top level parents skipped for some reason: not contained in either `isBlended` or `isIsolated`.
top level parents with only a single peak: `isIsolated`
source deblended from a parent with a single peak: `isIsolated`
source deblended from a parent with multiple peaks: `isBlended`

So the two are not complementary and we should keep the names, as they describe exactly what they contain. It would be possible to add another flag, `isBlend`, where `isBlend = (parent == 0) & ~isIsolated`, so that the entire source catalog would be the combination of the orthogonal flags `isBlended | isIolated | isBlend`. If you _really_ want me to I can add it, but our catalogs are already large enough and I don't see the point of adding a new flag that will never be used just to clarify `isBlended` and `isIsolated`. We should be able to do this in the docs, but obviously I haven't hit on the right explanation yet.

Reply:
I guess my main sticking point is in part the tense implied by `isBlended`.  Really, this represents a source that `wasBlended` or `isDeblendedChildOfBlend`, i.e. the "has been deblended from a parent with more than one child" nature just doesn't come across naturally to me with `isBlended`.  These suggestions might be (ok, _are_!) terrible for other reasons, so I will leave it to you if you want to change them but, as I mentioned below, I don't think I'm the only one who will be confused...as I mentioned below, just the first line in the doc jars with the name (`isBlended` --> "...source that was deblended...").  At the very least, could you add those double-`False` cases to the docs (prolly for `isBlended`)...I totally agree we don't want to add yet another flag column!